config/zookeeper_persister_test.go

```diff
@@ -179,14 +175,60 @@ func TestHistoricalConfigs(t *testing.T) {
 		t.Fatalf("Historical configs are not correct: %+v", cfgs)
 	}
 
-	newConfig, err := Unmarshal(cfgs[0])
-	helpers.CheckError(t, err)
+	newConfig := unmarshallOrPanic(cfgs[0])
 
 	if newConfig.Namespaces["test"] == nil {
 		t.Errorf("Config is not valid: %+v", newConfig)
 	}
 }
 
+func TestReadingStaleVersions(t *testing.T) {
+	conn, _, err := zk.Connect(servers, sessionTimeout)
+	helpers.PanicError(err)
+
+	defer conn.Close()
+
+	p, err := NewZkConfigPersister("/conflicting", servers)
+	helpers.CheckError(t, err)
+
+	defer p.(*ZkConfigPersister).Close()
+
+	origCfg := NewDefaultServiceConfig()
+	origCfg.Namespaces["test"] = NewDefaultNamespaceConfig("test")
+	origCfg.Version = 200
+
+	persistOrPanic(p, origCfg)
+
+	// Two changes
+	cfg1 := cloneConfig(origCfg)
+	cfg1.Namespaces["test_new"] = NewDefaultNamespaceConfig("test_new")
+	cfg1.Version = origCfg.Version + 1
+
+	cfg2 := cloneConfig(origCfg)
+	cfg2.Namespaces["test_newer"] = NewDefaultNamespaceConfig("test_newer")
+	cfg2.Version = origCfg.Version + 2
+
+	// Persist in quick succession
```

Comment:
Could this end up being flaky eventually?

Reply:
Yes. But this is temporary - working on a better test + a fix. I could leave out this test if you feel strongly about it.